### DISCUSSION TITLE
Deno adapter custom response headers for static files

### DISCUSSION THREAD

**Discussion by crispinb** [withastro/roadmap#486] [Proposal]
### Body

# Summary

The Deno adapter should by some means enable customising responses (particularly headers, but perhaps more) for statically-served files

# Background & Motivation

This arose from wanting to host a personal website built with Astro on Deno Deploy. Static images are all served without cache headers, Deno Deploy doesn't have general http server knobs to twiddle, and the Astro Deno adapter serves static files without intervention. Clients must therefore download static assets anew on every request.

# Goals

- enable cache control for static assets by adding custom response headers to http responses serving static files
- in the most minimal form this might take, these could be provided in the adapter options, mapping 
  from file type/extension to headers
- a more generalised approach is also suggested below

# Example

I've edited the Astro Deno adapter for my own use such that it accepts a 'headerMap' option, configured in `astro.config.mjs` like so:

```js
  adapter: deno({
       headerMap: { "png": ["cache-control", "max-age:604800, immutable"]}
})
```

The adapter then adds those headers for each request for a file of the specified type.

# Possible generalisation
This might be a little over-specific to my use case for a general purpose module like the adapter. 
A more flexible alternative might be to integrate the [Oak middleware framework](https://oakserver.github.io/oak/) into the adapter. The options bag could pass specifications for middleware to be injected per Deno response.  

I haven't needed this for my very simple personal site though I'm considering it for a more adequate caching implementation (ie. with cache busting).  I'd be happy to put together a POC for doing it this way if there's any interest.

**Comment by florian-lefebvre** [comment]
The deno adapter is now maintained by the Deno team: https://docs.astro.build/en/guides/integrations-guide/deno/
